ballerina/tests/35_subscriptions.bal

```diff
@@ -150,6 +138,30 @@ isolated function testSubscriptionWithVariables() returns error? {
     }
 }
 
+@test:Config {
+    groups: ["introspection", "typename", "subscriptions"]
+}
+isolated function testSubscriptionWithIntrsopectionInFields() returns error? {
+    string document = string`subscription { students { __typename } }`;
+    string url = "ws://localhost:9091/subscriptions";
+    websocket:Client wsClient = check new(url);
+    check writeWebSocketTextMessage(document, wsClient);
+    json expectedPayload = { data: { students: { __typename:"StudentService" }}};
+    check validateWebSocketResponse(wsClient, expectedPayload);
+}
+
+@test:Config {
+    groups: ["subscriptions"]
+}
+isolated function testInvalidSubscription() returns error? {
+    string document = string`subscription { invalidField }`;
+    string url = "ws://localhost:9091/subscriptions";
+    websocket:Client wsClient = check new(url);
+    check writeWebSocketTextMessage(document, wsClient);
+    json expectedPayload = check getJsonContentFromFile("subscription_invalid_field.json");
+    check validateWebSocketResponse(wsClient, expectedPayload);
+}
+
```

Comment:
Do we have tests for invalid WebSocket payload validation?

Reply:
Currently, no. I'll add them.